### PR TITLE
Add new camp and umpire versions

### PR DESCRIPTION
**Pull request by white238**
This works

### CODE REVIEW

**Review by bmhan12 — APPROVED** [review]

File: packages/camp/package.py [hunk 1]
```diff
@@ -137,6 +137,7 @@ class Camp(CMakePackage, CudaPackage, ROCmPackage):
     maintainers = ["trws"]
 
     version("main", branch="main", submodules="False")
+    version("2023.06.0", sha256="b828f650727717578317d7e048048b279a2670bb82a0f0aaf3196fefb0bafca9")
```
Comment (rhornung67) [inline packages/camp/package.py:140]: which commit does this sha hash refer to? There is a camp tag v2023.06.0.

File: packages/umpire/package.py [hunk 2]
```diff
@@ -115,6 +116,7 @@ class Umpire(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on("camp", when="@5.0.0:")
     depends_on("camp@0.2.2:0.2.3", when="@6.0.0")
     depends_on("camp@0.1.0", when="@5.0.0:5.0.1")
+    depends_on("camp@2023.06", when="@2023.06")
```
Comment (rhornung67) [inline packages/umpire/package.py:119]:
```suggestion
    depends_on("camp@2023.06.0", when="@2023.06.0")
```

Comment (white238) [reply packages/umpire/package.py:119]: It matches any patch version if you don't specify it.

File: packages/camp/package.py (hunk 1 above)
Comment (white238) [reply packages/camp/package.py:140]: It is the commit of the release. I was following the pattern in the file.

Comment (rhornung67) [reply packages/camp/package.py]: Not sure why it is using commit hashes, each camp release is tagged.

**Pull request merged**